The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12724643901/job/35471017216) [(plain)](https://github.com/rust-lang/rust/commit/fe921ed3f3cd6a328af76d1be1ab7b74b77e1857/checks/35471017216/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#21 exporting to docker image format
#21 sending tarball 27.3s done
#21 DONE 40.6s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
To only update this specific test, also pass `--test-args consts/min_const_fn/bad_const_fn_body_ice.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/consts/min_const_fn/bad_const_fn_body_ice.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/min_const_fn/bad_const_fn_body_ice" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error[E0015]: cannot call non-const associated function `Box::<[i32; 3]>::new` in constant functions
##[error]  --> /checkout/tests/ui/consts/min_const_fn/bad_const_fn_body_ice.rs:2:5
   |
---
------------------------------------------


---- [ui] tests/ui/error-codes/E0010-teach.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0010-teach/E0010-teach.stderr"

- error[E0010]: allocations are not allowed in constants
+ error[E0015]: cannot call non-const associated function `Box::<[i32; 3]>::new` in constants
2   --> $DIR/E0010-teach.rs:5:23
2   --> $DIR/E0010-teach.rs:5:23
3    |
4 LL | const CON: Vec<i32> = vec![1, 2, 3];
-    |                       ^^^^^^^^^^^^^ allocation not allowed in constants
+    |                       ^^^^^^^^^^^^^
6    |
-    = note: The runtime heap is not yet available at compile-time, so no runtime heap allocations can be created.
---
To only update this specific test, also pass `--test-args error-codes/E0010-teach.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/error-codes/E0010-teach.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0010-teach" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-Z" "teach"
--- stderr -------------------------------
error[E0015]: cannot call non-const associated function `Box::<[i32; 3]>::new` in constants
##[error]  --> /checkout/tests/ui/error-codes/E0010-teach.rs:5:23
   |
   |
LL | const CON: Vec<i32> = vec![1, 2, 3]; //~ ERROR E0010
   |
   = note: calls in constants are limited to constant functions, tuple structs and tuple variants
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)


error[E0015]: cannot call non-const method `slice::<impl [i32]>::into_vec::<std::alloc::Global>` in constants
##[error]  --> /checkout/tests/ui/error-codes/E0010-teach.rs:5:23
   |
LL | const CON: Vec<i32> = vec![1, 2, 3]; //~ ERROR E0010
   |
   = note: calls in constants are limited to constant functions, tuple structs and tuple variants
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

---
- error[E0010]: allocations are not allowed in constants
+ error[E0015]: cannot call non-const associated function `Box::<[i32; 3]>::new` in constants
2   --> $DIR/E0010.rs:3:23
3    |
4 LL | const CON: Vec<i32> = vec![1, 2, 3];
-    |                       ^^^^^^^^^^^^^ allocation not allowed in constants
+    |                       ^^^^^^^^^^^^^
6    |
+    = note: calls in constants are limited to constant functions, tuple structs and tuple variants
---
To only update this specific test, also pass `--test-args error-codes/E0010.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/error-codes/E0010.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0010" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error[E0015]: cannot call non-const associated function `Box::<[i32; 3]>::new` in constants
##[error]  --> /checkout/tests/ui/error-codes/E0010.rs:3:23
   |
   |
LL | const CON: Vec<i32> = vec![1, 2, 3]; //~ ERROR E0010
   |
   = note: calls in constants are limited to constant functions, tuple structs and tuple variants
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)


error[E0015]: cannot call non-const method `slice::<impl [i32]>::into_vec::<std::alloc::Global>` in constants
##[error]  --> /checkout/tests/ui/error-codes/E0010.rs:3:23
   |
LL | const CON: Vec<i32> = vec![1, 2, 3]; //~ ERROR E0010
   |
   = note: calls in constants are limited to constant functions, tuple structs and tuple variants
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

---
34 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/macros/vec-macro-in-pattern.rs:7:14
- LL |         Some(vec![43]) => {} //~ ERROR expected a pattern, found a function call
+ error[E0164]: expected tuple struct or tuple variant, found associated function `::alloc::boxed::Box::new`
+    |
+ LL |         Some(vec![43]) => {}
+    |              ^^^^^^^^ `fn` calls are not allowed in patterns
+    |
---
To only update this specific test, also pass `--test-args macros/vec-macro-in-pattern.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/macros/vec-macro-in-pattern.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/macros/vec-macro-in-pattern" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error[E0658]: usage of qualified paths in this context is experimental
##[error]  --> /checkout/tests/ui/macros/vec-macro-in-pattern.rs:7:14
   |
   |
LL |         Some(vec![43]) => {} //~ ERROR expected a pattern, found a function call
   |
   = note: see issue #86935 <https://github.com/rust-lang/rust/issues/86935> for more information
   = help: add `#![feature(more_qualified_paths)]` to the crate attributes to enable
   = help: add `#![feature(more_qualified_paths)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0164]: expected tuple struct or tuple variant, found associated function `<[_]>::into_vec`
   |
   |
LL |         Some(vec![43]) => {} //~ ERROR expected a pattern, found a function call
   |              ^^^^^^^^ `fn` calls are not allowed in patterns
   = help: for more information, visit https://doc.rust-lang.org/book/ch19-00-patterns.html
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0164]: expected tuple struct or tuple variant, found associated function `::alloc::boxed::Box::new`
error[E0164]: expected tuple struct or tuple variant, found associated function `::alloc::boxed::Box::new`
##[error]  --> /checkout/tests/ui/macros/vec-macro-in-pattern.rs:7:14
   |
LL |         Some(vec![43]) => {} //~ ERROR expected a pattern, found a function call
   |              ^^^^^^^^ `fn` calls are not allowed in patterns
   = help: for more information, visit https://doc.rust-lang.org/book/ch19-00-patterns.html
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 3 previous errors
---

11 LL |   };
12    |   - value is dropped here
13 
- error[E0010]: allocations are not allowed in statics
+ error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
16    |
16    |
17 LL | static STATIC11: Vec<MyOwned> = vec![MyOwned];
-    |                                 ^^^^^^^^^^^^^ allocation not allowed in statics
+    |                                 ^^^^^^^^^^^^^
19    |
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
20    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
21 
22 error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics
38    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
38    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
39    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
- error[E0010]: allocations are not allowed in statics
- error[E0010]: allocations are not allowed in statics
+ error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
43    |
43    |
44 LL |     vec![MyOwned],
-    |     ^^^^^^^^^^^^^ allocation not allowed in statics
+    |     ^^^^^^^^^^^^^
46    |
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
47    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
48 
49 error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics

56    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
57    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
- error[E0010]: allocations are not allowed in statics
- error[E0010]: allocations are not allowed in statics
+ error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
61    |
61    |
62 LL |     vec![MyOwned],
-    |     ^^^^^^^^^^^^^ allocation not allowed in statics
+    |     ^^^^^^^^^^^^^
64    |
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
65    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
66 
67 error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics

74    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
75    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
- error[E0010]: allocations are not allowed in statics
- error[E0010]: allocations are not allowed in statics
+ error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
79    |
79    |
80 LL |     &vec![MyOwned],
-    |      ^^^^^^^^^^^^^ allocation not allowed in statics
+    |      ^^^^^^^^^^^^^
82    |
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
83    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
84 
85 error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics

92    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
93    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
- error[E0010]: allocations are not allowed in statics
- error[E0010]: allocations are not allowed in statics
+ error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
97    |
97    |
98 LL |     &vec![MyOwned],
-    |      ^^^^^^^^^^^^^ allocation not allowed in statics
+    |      ^^^^^^^^^^^^^
100    |
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
101    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
102 
103 error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics

110    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
111    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
- error[E0010]: allocations are not allowed in statics
- error[E0010]: allocations are not allowed in statics
+ error[E0015]: cannot call non-const associated function `Box::<[isize; 1]>::new` in statics
115    |
115    |
116 LL | static STATIC19: Vec<isize> = vec![3];
-    |                               ^^^^^^^ allocation not allowed in statics
+    |                               ^^^^^^^
118    |
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
119    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
120 
121 error[E0015]: cannot call non-const method `slice::<impl [isize]>::into_vec::<std::alloc::Global>` in statics

128    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
129    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
- error[E0010]: allocations are not allowed in statics
- error[E0010]: allocations are not allowed in statics
+ error[E0015]: cannot call non-const associated function `Box::<[isize; 1]>::new` in statics
133    |
133    |
134 LL |         static x: Vec<isize> = vec![3];
-    |                                ^^^^^^^ allocation not allowed in statics
+    |                                ^^^^^^^
136    |
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: calls in statics are limited to constant functions, tuple structs and tuple variants
+    = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
137    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
138 
139 error[E0015]: cannot call non-const method `slice::<impl [isize]>::into_vec::<std::alloc::Global>` in statics
163 
164 error: aborting due to 17 previous errors
165 
- Some errors have detailed explanations: E0010, E0015, E0493, E0507.
---
To only update this specific test, also pass `--test-args statics/check-values-constraints.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/statics/check-values-constraints.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/statics/check-values-constraints" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error[E0493]: destructor of `SafeStruct` cannot be evaluated at compile-time
##[error]  --> /checkout/tests/ui/statics/check-values-constraints.rs:64:7
   |
   |
LL |       ..SafeStruct {
   |  _______^
LL | |         //~^ ERROR destructor of
LL | |         field1: SafeEnum::Variant3(WithDtor),
LL | |         field2: SafeEnum::Variant1,
   | |_____^ the destructor for this type cannot be evaluated in statics
LL |   };
   |   - value is dropped here


error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
   |
   |
LL | static STATIC11: Vec<MyOwned> = vec![MyOwned];
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics
   |
   |
LL | static STATIC11: Vec<MyOwned> = vec![MyOwned];
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const method `<str as ToString>::to_string` in statics
   |
   |
LL |     field2: SafeEnum::Variant4("str".to_string()), //~ ERROR cannot call non-const method
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`

error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
   |
   |
LL |     vec![MyOwned], //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics
   |
   |
LL |     vec![MyOwned], //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
   |
   |
LL |     vec![MyOwned], //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics
   |
   |
LL |     vec![MyOwned], //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
   |
   |
LL |     &vec![MyOwned], //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics
   |
   |
LL |     &vec![MyOwned], //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const associated function `Box::<[MyOwned; 1]>::new` in statics
   |
   |
LL |     &vec![MyOwned], //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const method `slice::<impl [MyOwned]>::into_vec::<std::alloc::Global>` in statics
   |
   |
LL |     &vec![MyOwned], //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const associated function `Box::<[isize; 1]>::new` in statics
   |
   |
LL | static STATIC19: Vec<isize> = vec![3];
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const method `slice::<impl [isize]>::into_vec::<std::alloc::Global>` in statics
   |
   |
LL | static STATIC19: Vec<isize> = vec![3];
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const associated function `Box::<[isize; 1]>::new` in statics
   |
   |
LL |         static x: Vec<isize> = vec![3]; //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const method `slice::<impl [isize]>::into_vec::<std::alloc::Global>` in statics
   |
   |
LL |         static x: Vec<isize> = vec![3]; //~ ERROR allocations are not allowed in statics
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
   = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
error[E0507]: cannot move out of static item `x`
##[error]  --> /checkout/tests/ui/statics/check-values-constraints.rs:119:9
   |
LL |         x
LL |         x
   |         ^ move occurs because `x` has type `Vec<isize>`, which does not implement the `Copy` trait
help: consider borrowing here
   |
LL |         &x
   |         +
```

</details>

--- a/library/alloc/src/macros.rs
+++ b/library/alloc/src/macros.rs
@@ -48,8 +48,8 @@ macro_rules! vec {
     );
     ($($x:expr),+ $(,)?) => (
         <[_]>::into_vec(
-            // Using the intrinsic produces a dramatic improvement in compile
-            // time when constructing arrays with many elements.
+            // Using the intrinsic produces a dramatic improvement in stack usage for
+            // unoptimized programs using this code path to construct large Vecs.
             $crate::boxed::box_new([$($x),+])
         )
     );

